frontend/amundsen_application/static/js/pages/TableDetailPage/index.tsx

```diff
@@ -223,6 +233,19 @@ export class TableDetail extends React.Component<
     }
   };
 
+  handleExpandCollapseAllBtnVisibility = () => {
+    const { isRightPanelOpen } = this.state;
+    const minWidth = isRightPanelOpen
+      ? Constants.MIN_WIDTH_DISPLAY_BTN_WITH_OPEN_PANEL
+      : Constants.MIN_WIDTH_DISPLAY_BTN;
+
+    if (window.matchMedia(`(min-width: ${minWidth}px)`).matches) {
+      this.setState({ isExpandCollapseAllBtnVisible: true });
+    } else {
+      this.setState({ isExpandCollapseAllBtnVisible: false });
+    }
```

Comment:
Nit, I would do:
```suggestion
    let newState = { isExpandCollapseAllBtnVisible: false };
    
    if (window.matchMedia(`(min-width: ${minWidth}px)`).matches) {
      newState = { isExpandCollapseAllBtnVisible: true };
    }
    this.setState(newState);
```
So we asses the new state, execute it only once. I also saves one branch!